### PR TITLE
Maverick DEX Trades

### PR DESCRIPTION
**Pull request by gte620v**
Brief comments on the purpose of your changes:

Adding trades for Maverick, app.mav.xyz

**For Dune Engine V2**

I've checked that:

### General checks:
* [x] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [x] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [x] if adding a new model, I added a test
* [x] the filename is unique and ends with .sql
* [x] each sql file is a select statement and has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`
* [x] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [x] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [x] `coin_id` represents the ID of the coin on coinpaprika.com
* [x] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [x] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [x] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [x] where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Comment by gte620v** [comment]
Maverick is a new DEX.  UI is at https://app.mav.xyz/pools

**Comment by jeff-dude** [comment]
looks like the model runs fine, but we will want to add the seed row(s) back to ensure the test is passing. i'd say try again with adding it back, be extra careful on formatting/commas/etc.

if there are still issues, @Hosuke will be able to help. he can also follow up with full review.

thanks for the patience and contribution!

**Comment by gte620v** [comment]
is there a way to run this locally? `dbt seed --select state:modified --state .` 

i get `Runtime Error
  Got a state selector method, but no comparison manifest`  when i run it.  and it seems that `aws s3 cp s3://manifest-spellbook/manifest.json manifest.json` is access controlled.

File: dbt_project.yml [hunk 1]
```diff
@@ -149,6 +149,11 @@ models:
       bnb:
         +schema: nomiswap_bnb
 
+    maverick:
+      +schema: maverick
+      ethereum:
+        +schema: ethereum
```
Comment (Hosuke) [inline dbt_project.yml:155]: It seems I cannot directly commit to this PR.
Please fix this:
```suggestion
        +schema: maverick_ethereum
```

**Comment by Hosuke** [comment]
GM @jeff-dude, [the dbt slim ci test failure](https://github.com/duneanalytics/spellbook/actions/runs/4388659794/jobs/7689643821#step:8:29) I actually have no idea, and this error happens in several PRs as well. We may need to debug for the root cause.

**Comment by gte620v** [comment]
Thanks @Hosuke I committed your change.  The CI passed.  

image

Although, I feel like it has passed before and then somehow it revert to a failed CI state.  Copying the screenshot to show that it was green post commit.

**Comment by Hosuke** [comment]
If we want to make maverick.trades shown up in `dex.trades`, we may still have some more steps to do:

1. Make a roll-up spell for `maverick.trades` (optional, but if there is other versions than `maverick_v1_ethereum.trades`, we can prepare this earlier). An example to follow: https://github.com/duneanalytics/spellbook/blob/main/models/balancer/balancer_trades.sql
2. Add `maverick.trades` to `dex.trades` meta-table: https://github.com/duneanalytics/spellbook/blob/main/models/dex/dex_trades.sql#L53

**Comment by gte620v** [comment]
@Hosuke Thanks for the pointer! 

I attempted the changes.  I am unsure of where I need `_v1` in the various `ref`s and where I don't.  Please let me know if I am misreferencing something and I will make the change.

File: models/maverick/maverick_trades.sql [hunk 2]
```diff
@@ -0,0 +1,48 @@
+{{ config(
+    schema = 'maverick',
+    alias = 'trades',
+    post_hook='{{ expose_spells(\'["arbitrum", "ethereum", "optimism", "polygon"]\',
```
Comment (Hosuke) [inline models/maverick/maverick_trades.sql]: For now may be just ethereum?
```suggestion
    post_hook='{{ expose_spells(\'["ethereum"]\',
```

Comment (gte620v) [reply models/maverick/maverick_trades.sql]: I think this is fixed.  It isn't letting me commit this suggestion because it is already there.

@Hosuke

**Comment by Hosuke** [comment]
> @Hosuke Thanks for the pointer!
> 
> I attempted the changes. I am unsure of where I need `_v1` in the various `ref`s and where I don't. Please let me know if I am misreferencing something and I will make the change.

I have checked the changes and I think you have done ref everywhere right.

File: models/maverick/ethereum/maverick_v1_ethereum_trades.sql [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+WITH dexs AS
+(
+    --Maverick V1
+    SELECT
+        t.evt_block_time AS block_time
+        ,t.recipient AS taker
+        ,'' AS maker
+        ,amountOut AS token_bought_amount_raw
+        ,amountIn AS token_sold_amount_raw
+        ,NULL AS amount_usd
```
Comment (Hosuke) [inline models/maverick/ethereum/maverick_v1_ethereum_trades.sql:27]:
```suggestion
        ,CAST(NULL AS DOUBLE) AS amount_usd
```

File: models/maverick/maverick_trades.sql [hunk 4]
```diff
@@ -0,0 +1,48 @@
+{{ config(
+    schema = 'maverick',
```
Comment (Hosuke) [inline models/maverick/maverick_trades.sql:2]: This line can be removed as the schema already defined in `dbt_project.yml`
```suggestion
```

**Comment by gte620v** [comment]
@Hosuke looks like the test is failing after committing the suggested fixes.  any ideas?

**Comment by gte620v** [comment]
@Hosuke I gave you write rights to the fork.  Feel free to commit directly.

**Comment by Hosuke** [comment]
> Details

It's due to data freshness, and it is unrelated to this PR.

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @gte620v 

[Sample maverick v1 ethereum trades](https://dune.com/queries/2202651?d=1)

**Pull request closed without merging**